aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -93,146 +165,126 @@ module AptosFramework::Token {
     }
 
-    /// The data associated with the Tokens
-    struct TokenData has copy, drop, store {
-        // Unique name within this creators account for this Token's collection
-        collection: ASCII::String,
-        // Describes this Token
-        description: ASCII::String,
-        // The name of this Token
-        name: ASCII::String,
-        // Optional maximum number of this type of Token.
-        maximum: Option<u64>,
-        // Total number of this type of Token
-        supply: Option<u64>,
-        // URL for additional information / media
-        uri: ASCII::String,
-        // the royalty of the token
-        royalty: Royalty,
+    /// create collection event with creator address and collection name
+    struct CreateCollectionEvent has drop, store {
+        creator: address,
+        collection_name: String,
+        uri: String,
+        description: String,
+        maximum: u64
     }
 
     /// Capability required to mint tokens.
     struct MintCapability has store {
-        token_id: TokenId,
+        token_id: TokenId
     }
 
     /// Capability required to burn tokens.
     struct BurnCapability has store {
-        token_id: TokenId,
+        token_id: TokenId
     }
 
     //
     // Creator Script functions
     //
 
-    public(script) fun create_limited_collection_script(
+    public(script) fun create_collection_script(
         creator: &signer,
         name: vector<u8>,
         description: vector<u8>,
         uri: vector<u8>,
         maximum: u64,
+        mutate_setting: vector<bool>
     ) acquires Collections {
         create_collection(
             creator,
             ASCII::string(name),
             ASCII::string(description),
             ASCII::string(uri),
-            Option::some(maximum),
-        );
-    }
-
-    public(script) fun create_unlimited_collection_script(
-        creator: &signer,
-        name: vector<u8>,
-        description: vector<u8>,
-        uri: vector<u8>,
-    ) acquires Collections {
-        create_collection(
-            creator,
-            ASCII::string(name),
-            ASCII::string(description),
-            ASCII::string(uri),
-            Option::none(),
+            maximum,
+            mutate_setting
         );
     }
 
-    public(script) fun create_limited_token_script(
+    public(script) fun create_token_script(
         creator: &signer,
         collection: vector<u8>,
         name: vector<u8>,
         description: vector<u8>,
-        monitor_supply: bool,
-        initial_balance: u64,
+        balance: u64,
         maximum: u64,
         uri: vector<u8>,
-        royalty_points_per_million: u64
+        royalty_payee_address: address,
+        royalty_points_denominator: u64,
+        royalty_points_nominator: u64,
+        token_mutate_setting: vector<bool>,
+        property_keys: vector<String>,
+        property_values: vector<vector<u8>>,
+        property_types: vector<String>
     ) acquires Collections, TokenStore {
```

Comment:
I wonder if we should keep our legacy API for a release or two and have a new API that is more encompassing. That way you don't have to update every single aspect of the tokens before landing this.

Reply:
as discussed offline, this change will be only be in tokenV1. TokenV1 and Token will co-exist 1-2 weeks to update the dependencies in SDK/Indexer and also notify community to update.